magenta/music/musicxml_parser.py

```diff
@@ -0,0 +1,678 @@
+    self.pitch = (pitch_string, midi_pitch)
+
+  def parsetuplet(self, xml_time_modification):
+    """Parses a tuplet ratio, represented in MusicXML by the
+    <time-modification> element
+    """
+    numerator = int(xml_time_modification.find("actual-notes").text)
+    denominator = int(xml_time_modification.find("normal-notes").text)
+    self.note_duration.tuplet_ratio = Fraction(numerator, denominator)
+
+  @staticmethod
+  def pitchtomidipitch(step, alter, octave):
+    """Convert MusicXML pitch representation to MIDI pitch number"""
+    pitch_class = 0
+    if step == "C":
+      pitch_class = 0
+    elif step == "D":
+      pitch_class = 2
+    elif step == "E":
+      pitch_class = 4
+    elif step == "F":
+      pitch_class = 5
+    elif step == "G":
+      pitch_class = 7
+    elif step == "A":
+      pitch_class = 9
+    elif step == "B":
+      pitch_class = 11
+
+    pitch_class = (pitch_class + int(alter)) % 12
+    midi_pitch = (12 + pitch_class) + (int(octave) * 12)
+    return midi_pitch
+
+  def __str__(self):
+    note_string = "{duration: " + str(self.note_duration.duration)
+    note_string += ", midi_ticks: " + str(self.note_duration.midi_ticks)
+    note_string += ", seconds: " + str(self.note_duration.seconds)
+    if self.is_rest:
+      note_string += ", rest: " + str(self.is_rest)
+    else:
+      note_string += ", pitch: " + self.pitch[0]
+      note_string += ", MIDI pitch: " + str(self.pitch[1])
+
+    note_string += ", voice: " + str(self.voice)
+    note_string += ", velocity: " + str(self.velocity) + "} "
+    note_string += "(@time: " + str(self.note_duration.time_position) + ")"
+    return note_string
+
+
+class NoteDuration(object):
+  """Internal representation of a MusicXML note's duration properties"""
+  def __init__(self, state):
+    self.duration = 0                   # MusicXML duration
+    self.midi_ticks = 0                 # Duration in MIDI ticks
+    self.seconds = 0                    # Duration in seconds
+    self.time_position = 0              # Onset time in seconds
+    self.dots = 0                       # Number of augmentation dots
+    self.type = "quarter"               # MusicXML duration type
+    self.tuplet_ratio = Fraction(1, 1)  # Ratio for tuplets (default to 1)
+    self.state = state
+
+  def parseduration(self, note, duration):
+    """Parse the duration of a note and compute timings"""
+    self.duration = int(duration)
+
+    self.midi_ticks = self.duration
+    self.midi_ticks *= (constants.STANDARD_PPQ / self.state.divisions)
+
+    self.seconds = (self.midi_ticks / constants.STANDARD_PPQ)
+    self.seconds *= self.state.seconds_per_beat
+
+    self.time_position = self.state.time_position
+
+    if note.is_in_chord:
+      # If this is a chord, set the time position to the time position
+      # of the previous note (i.e. all the notes in the chord will have
+      # the same time position)
+      self.time_position = self.state.previous_note.time_position
```

Comment:
When I tried to run convert_musicxml_dir_to_note_sequences on the test xml files, I got an error caused by this line:

```
...
  File ".../magenta/bazel-bin/magenta/scripts/convert_musicxml_dir_to_note_sequences.runfiles/__main__/magenta/music/musicxml_parser.py", line 528, in parseduration
    self.time_position = self.state.previous_note.time_position
AttributeError: 'Note' object has no attribute 'time_position'
```

Do you know why the unit tests didn't catch this?